### PR TITLE
EVG-18225: Check status code and report error while getting signed ur…

### PR DESCRIPTION
**Pull request by mridul-augustine**
…l in poplar

### CODE REVIEW

File: rpc/client.go [hunk 1]
```diff
@@ -198,6 +198,13 @@ func getSignedURL(opts *UploadReportOptions) (string, error) {
 	}
 
 	defer response.Body.Close()
+	grip.Debug(message.Fields{
+		"Status Code": response.StatusCode,
+	})
+	if response.StatusCode != http.StatusOK {
+		return "", errors.Errorf("Non-OK HTTP status. Status Code: %d. Status Text: %s", response.StatusCode, http.StatusText(response.StatusCode))
```
Comment (julianedwards) [inline rpc/client.go:205]: Can you change this to `"failed to get the Data-Pipes signed URL with response '%s'", response.Status`.

File: rpc/client.go [hunk 2]
```diff
@@ -198,6 +198,13 @@ func getSignedURL(opts *UploadReportOptions) (string, error) {
 	}
 
 	defer response.Body.Close()
+	grip.Debug(message.Fields{
+		"Status Code": response.StatusCode,
```
Comment (julianedwards) [inline rpc/client.go:202]: I would add more context to this message.

File: rpc/client.go [hunk 3]
```diff
@@ -225,6 +232,14 @@ func uploadTestReport(signedURL string, data []byte, client *http.Client) error
 		return errors.Wrap(err, "report upload to given signed URL")
 	}
 
+	defer response.Body.Close()
+	grip.Debug(message.Fields{
+		"Status Code": response.StatusCode,
+	})
+	if response.StatusCode != http.StatusOK {
+		return errors.Errorf("Non-OK HTTP status. Status Code: %d. Status Text: %s", response.StatusCode, http.StatusText(response.StatusCode))
```
Comment (julianedwards) [inline rpc/client.go:240]: Can you change this to `"failed to upload test report to the Data-Pipes service with response '%s'", response.Status`.

File: rpc/client.go [hunk 4]
```diff
@@ -198,6 +198,13 @@ func getSignedURL(opts *UploadReportOptions) (string, error) {
...
+	grip.Debug(message.Fields{
+		"Status Code": response.StatusCode,
+	})
+	if response.StatusCode != http.StatusOK {
+		return "", errors.Errorf("Non-OK HTTP status. Status Code: %d. Status Text: %s", response.StatusCode, http.StatusText(response.StatusCode))
+	}
+
 	body, error := ioutil.ReadAll(response.Body)
 	if error != nil {
 		return "", err
```
Comment (julianedwards) [inline rpc/client.go:210]: I would wrap this error.

File: rpc/client.go [hunk 5]
```diff
@@ -198,6 +198,13 @@ func getSignedURL(opts *UploadReportOptions) (string, error) {
...
 
 	defer response.Body.Close()
+	grip.Debug(message.Fields{
+		"Status Code": response.StatusCode,
+	})
+	if response.StatusCode != http.StatusOK {
+		return "", errors.Errorf("Non-OK HTTP status. Status Code: %d. Status Text: %s", response.StatusCode, http.StatusText(response.StatusCode))
+	}
+
 	body, error := ioutil.ReadAll(response.Body)
```
Comment (julianedwards) [inline rpc/client.go:208]: `err`

File: rpc/client.go (hunk 4 above)
Comment (julianedwards) [reply rpc/client.go:210]: Also, on L214 we aren't properly checking the error, you can change it to:
```
if err = json.Unmarshal(body, &responseBody); err != nil {
        return "", errors.Wrap(err, "parsing signed URL response")
}
```

File: rpc/client.go [hunk 6]
```diff
@@ -198,6 +198,13 @@ func getSignedURL(opts *UploadReportOptions) (string, error) {
...
 	if error != nil {
```
Comment (julianedwards) [inline rpc/client.go:209]: `if err`

**Review by julianedwards — CHANGES_REQUESTED** [review]

**Review by julianedwards — CHANGES_REQUESTED** [review]
I think all of the debug logs are a bit overkill, but if you find them useful feel free to keep them.

**Comment by mridul-augustine** [comment]
> I think all of the debug logs are a bit overkill, but if you find them useful feel free to keep them.

I just added one debug :) others were just the old ones. Just added more context

**Review by julianedwards — CHANGES_REQUESTED** [review]
A few more things!

**Review by julianedwards — APPROVED** [review]

**Pull request merged**